Allow templating mqtt fan speed command value

dgwynne opened the pull request:
I have a pedestal fan that has eight different speeds which are
shown on the front of the unit as 1 through 8. I'm running tasmota
on the fan, and the values for the speeds "on the wire" are 0 through
7. I want to present the speeds in home-assistant as they are on
the front of the unit, so I need to be able to transform the value
coming from the state topic, but also for the command topic.

Worse, this device uses a Tuya MCU, which tasmota has very green
support for. To set the speed I need to use the TuyaSend4 command,
which has a payload that has to be prefixed with the DpId, which
again, is much easier if I can template the command value.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
fan:
  - platform: mqtt
    name: "Anko HEGSM40"
    state_topic: "stat/anko_fan/POWER1"
    command_topic: "cmnd/anko_fan/POWER1"
    oscillation_state_topic: "stat/anko_fan/POWER2"
    oscillation_command_topic: "cmnd/anko_fan/POWER2"
    speed_state_topic: "stat/anko_fan/SPEED"
    speed_value_template: "{{ value.split(',')[1] | int + 1 }}"
    speed_command_topic: "cmnd/anko_fan/TuyaSend4"
    speed_command_template: "3,{{ value | int - 1 }}"
    qos: 0
    payload_on: 'ON'
    payload_off: 'OFF'
    payload_oscillation_on: 'ON'
    payload_oscillation_off: 'OFF'
    availability_topic: tele/anko_fan/LWT
    payload_available: Online
    payload_not_available: Offline
    speeds:
      - '1'
      - '2'
      - '3'
      - '4'
      - '5'
      - '6'
      - '7'
      - '8'
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

balloob committed:
Fix automation sun import (#31521)

chmielowiec committed:
Fix migrating huawei_lte entry without recipient (#31522)

frenck committed:
Limit OAuth scopes for Netatmo and Home Assistant Cloud (#31538)

* Limit OAuth scopes for Netatmo and Home Assistant Cloud

* Fix tests by making order of scopes predictable

d0ugal committed:
Only normalise Garmin connect data to minutes if the value is not None (#31526)

Otherwise this causes additional TypeError messages to be logged for
division of None.

Quentame committed:
Fix iCloud determine_interval: add default interval to max_interval (#31533)

balloob committed:
Deprecate old netatmo keys (#31544)

bendavid committed:
update aiopylgtv to 0.3.3 (#31545)

scop committed:
Use min and m as units in Garmin Connect for consistency and correctness (#31543)

balloob committed:
Guard for reloading with no zone config (#31547)

balloob committed:
Bumped version to 0.105.2

balloob committed:
Update MQTT service description

balloob committed:
Merge pull request #31550 from home-assistant/rc

0.105.2

Adminiuga committed:
Bump ZHA dependencies. (#31555)

frenck committed:
Bump adguardhome to 0.4.1 (#31565)

engrbm87 committed:
Fix librouteros response error handling (#31588)

MatthewFlamm committed:
update pynws to 0.10.4 (#31591)

Adminiuga committed:
Bump ZHA dependencies. (#31619)

Bump up zigpy-homeassistant==0.13.2

Danielhiversen committed:
Fix hvac_action for mill (#31630)

cyberjunky committed:
Catch garmin_connect keyerrors with unknown entity type updates (#31608)

* Catch keyerrors with unknown entity type updates

* Change debug level and removed . from log call

bdraco committed:
Resolve August integration makes too many requests and hits rate limits (#31558)

balloob committed:
Guard writing automation/scene/script config (#31568)

dgwynne committed:
Allow templating of the value used with the fan speed command.

I have a pedestal fan that has eight different speeds which are
shown on the front of the unit as 1 through 8. I'm running tasmota
on the fan, and the values for the speeds "on the wire" are 0 through
7. I want to present the speeds in home-assistant as they are on
the front of the unit, so I need to be able to transform the value
coming from the state topic, but also for the command topic.

Worse, this device uses a Tuya MCU, which tasmota has very green
support for. To set the speed I need to use the TuyaSend4 command,
which has a payload that has to be prefixed with the DpId, which
again, is much easier if I can template the command value.

dgwynne committed:
Allow templating of the value used with the fan speed command.

I have a pedestal fan that has eight different speeds which are
shown on the front of the unit as 1 through 8. I'm running tasmota
on the fan, and the values for the speeds "on the wire" are 0 through
7. I want to present the speeds in home-assistant as they are on
the front of the unit, so I need to be able to transform the value
coming from the state topic, but also for the command topic.

Worse, this device uses a Tuya MCU, which tasmota has very green
support for. To set the speed I need to use the TuyaSend4 command,
which has a payload that has to be prefixed with the DpId, which
again, is much easier if I can template the command value.

homeassistant commented:
Hello @dgwynne,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`081929a3f59b108c8dd6a1c1da989a8f6f580e70`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;31655&#x2F;commits&#x2F;081929a3f59b108c8dd6a1c1da989a8f6f580e70) - This commit has something that looks like an email address (david@gwynne.id.au). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

dgwynne commented:
I've updated my email address settings in GitHub, so the commit should be able to be linked to my GH account now.

homeassistant commented:
Hi @dgwynne, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#31655).

Once you do that we will be able to review and accept this pull request.

Thanks!

dgwynne commented:
I've signed the CLA.

crallian committed:
Added zone type Technical as power. (#31611)

*The zone type technical can be used in SPC to track status of e.g. mapping keys and outputs.

TechnicallyJoe committed:
Add recorder vars db_max_retries and db_retry_wait (#31561)

* added recorder vars db_max_retries and db_retry_wait

* fixed test_recorder_setup_failure

I failed because it was missing the two new variables. I simply added these with default values.

* fixed syntax error in test_recorder_setup_failure

* fixed formatting error in test_init_py for recorder component

* fixed typo in test case

* Updated the way the default keys for db_,max_wait and db_retry_wait is set

Implemented based on suggestions from @springstan

* Updated config_schema call to adhere to Black

* changed conf.get to conf[dict] for var retrieval

* removed 2 blank lines

benleb committed:
Sure Petcare new features various improvements (#31437)

* add typing

* 100% battery_level is enough

* human-friendly datetime

* better enum usage

* add online and learning mode attrs

* use max two decimals in attrs

* use legacy style debug logging

* remove str usage of enums

* add feeder

* add feeder and adapt to new surepy version

* use ProductID instead of ThingID

* various changes and improvements

* add connectivity sensors for all devices & proper support for multiple hubs

* remove "side effects"/exception catching in attribs

* correct unique ids, reorder classes

* move Flap class from binary_sensor to sensor and add a sensore base class

* comments cleanup, minor typing and logging fixes

* remove commented code

* remove commented code

* add typing

* 100% battery_level is enough

* human-friendly datetime

* better enum usage

* add online and learning mode attrs

* use max two decimals in attrs

* use legacy style debug logging

* remove str usage of enums

* add feeder

* add feeder and adapt to new surepy version

* use ProductID instead of ThingID

* various changes and improvements

* add connectivity sensors for all devices & proper support for multiple hubs

* remove "side effects"/exception catching in attribs

* correct unique ids, reorder classes

* move Flap class from binary_sensor to sensor and add a sensore base class

* comments cleanup, minor typing and logging fixes

* remove commented code

* remove commented code

* fix spelling in comment to make the CI happy (seriously?!)

* fix manifest file

* fix requirements_all.txt file

* add missing docstrings

* fix available property

* remove typing from self

* remove commented code

* remove is_on property from sensor

* jump to new surepy version

* remove useles init methods

vcc-sanya committed:
Add ZHA Texas Instruments CC device support (#31621)

homeassistant committed:
[ci skip] Translation update

Adminiuga committed:
ZHA tests refactoring (#31682)

* Fixtures for restoring/joning a device.

* binary_sensor.zha tests.

* cover.zha platform tests.

* device_tracker.zha platform tests.

* fan.zha platform tests.

* switch.zha platform tests.

* Update light.zha platform tests.

* Update sensor.zha platform tests.

* ZHA api tests refactoring.

* Update lock.zha platform tests.

* Update ZHA gateway tests.

* Update zha device action tests.

* Update zha device trigger tests.

* Cleanup.

balloob committed:
Clean up frontend services and events (#31654)

* Clean up frontend services and events

* Fix bug in core instead

* Add test that core works correctly with callback marked async funcs

balloob committed:
Add brightness light device actions (#31567)

cgtobi committed:
Fix wrong error message in netatmo integration (#31690)

cgtobi committed:
Fix wrong error message in netatmo integration (#31690)

Quentame committed:
Fix Météo-France I/O while testing (#31695)

* Fix Météo-France I/O while testing

* Review

cereal2nd committed:
Bump velbus version + load velbus module names into device info (#31664)

oischinger committed:
Vicare water_heater set_temperature fix and bump PyVicare to 0.1.7 (#31672)

* Fix ViCare water_heater set_temperature

This fixes a obvious but undiscovered bug in the water heater component:
Instead of the commanded value the prvious value was set on the API

* Bump PyVicare to 0.1.7

Commit:
Add MELCloud integration (#30712)

* Add MELCloud integration

* Provides a climate and sensor platforms. Multiple platforms on one go
is not the best option, but it does not make sense to remove them and
commit them later either.

* Email and access token are stored to the ConfigEntry. The token can be
updated by adding the integration again with the same email address. The
config flow is aborted and the update is performed on the background.

* Run isort

* Fix pylint errors

* Run black

* Increase coverage

* Update pymelcloud dependency

* Add HVAC_MODE_OFF emulation

* Remove print

* Update pymelcloud to enable device type filtering

* Collapse except blocks and chain ClientNotReadys

* Add preliminary documentation URL

* Use list comp for creating model info

Filters out empty model names form units.

* f-string galore

Dropped 'HVAC' from AtaDevice name template.

* Delegate fan mode mapping to pymelcloud

* Fix type annotation

* Access AtaDevice through self._device

* Prefer list comprehension

* Update pymelcloud to leverage device type grouping

The updated backend lib returns devices in a dict grouped by the device
type. The devices do not necessarily need to be in a single list and
this way isinstance is not required to extract devices by type.

* Remove DOMAIN presence check

This does not seem to make much sense after all.

* Fix async_setup_entry

Entry setup used half-baked naming from few experimentations back.
The naming conventiens were unified to match the platforms.

A redundant noneness check was also removed after evaluating the
possible return values from the backend lib.

* Simplify empty model name check

* Improve config validation

* Use config_validation strings.

* Add CONF_EMAIL to config schema. The value is not strictly required
when configuring through configuration.yaml, but having it there makes
things more consistent.

* Use dict[key] to access required properties.

* Add DOMAIN in config check back to async_setup. This is required if an
integration is configured throught config_flow.

* Remove unused manifest properties

* Remove redundant ClimateDevice property override

* Add __init__.py to coverage exclusion

* Use CONF_USERNAME instead of CONF_EMAIL

* Use asyncio.gather instead of asyncio.wait

* Misc fixes

* any -> Any

* Better names for dict iterations

* Proper dict access with mandatory/known keys

* Remove unused 'name' argument

* Remove unnecessary platform info from unique_ids

* Remove redundant methods from climate platform

* Remove redundant default value from dict get

* Update ConfigFlow sub-classing

* Define sensors in a dict instead of a list

* Use _abort_if_unique_id_configured to update token

* Fix them tests

* Remove current state guards

* Fix that gather call

* Implement sensor definitions without str manipulation

* Use relative intra-package imports

* Update homeassistant/components/melcloud/config_flow.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

ziv1234 committed:
Dynalite Integration (#27841)

* Initial commit

* ran hassfest and gen_requirements_all scripts

* fixes per request from Paulus Schoutsen

* ran gen_requirements_all

* updated library version - removed some debug leftover

* get_requirements again...

* added documentation URL

* ran isort

* changed storage in hass.data[DOMAIN] to use entry_id instead of host

* adopted unit tests to latest fix

* Update const.py

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

vzahradnik committed:
Extend Modbus binary sensor to support discrete inputs (#30004)

* Extend Modbus binary sensor to support discrete inputs

* Add backward compatibility for Modbus binary sensor

eavanvalkenburg committed:
Fix Evohome checking override duration (#31697)

balloob committed:
Bumped version to 0.105.3

raman325 committed:
For vizio integration, set unique ID early to prevent multiple zeroconf discovery items for the same device to appear (#31686)

* set unique ID early to prevent multiple zeroconf discovery items for the same device to appear

* add test

raman325 committed:
For vizio integration, set unique ID early to prevent multiple zeroconf discovery items for the same device to appear (#31686)

* set unique ID early to prevent multiple zeroconf discovery items for the same device to appear

* add test

SoftXperience committed:
Use latest version of python-pushover (forked) to fix issue with diff… (#31647)

* Use latest version of python-pushover (forked) to fix issue with different API tokens. (https://community.home-assistant.io/t/different-applications-in-pushover/6985)

* Rewrite pushover notify to use pushover_complete library

* Remove possibility to attach urls to notifications

* Fix comment

engrbm87 committed:
Add Glances sensors dynamically (#28639)

* Add temp_sensors to glances dynamically

* unsub from updates when sensor is disabled

* dynamicall add sensors

* change "device_name" to "mnt_point"

* remove unnecessary logging

* update sensor.py

* update test_config_flow.py

* remove commented code

balloob committed:
Merge pull request #31712 from home-assistant/rc

0.105.3

homeassistant committed:
[ci skip] Translation update

balloob committed:
Adjust entity slow warning for custom component (#31711)

balloob committed:
Limit derivative test (#31717)

balloob committed:
Fix person reload service (#31716)

balloob committed:
Fix person reload service (#31716)

gorynychzmey committed:
Add support of multiple Tado accounts (#31527)

* Added support of multiple Tado accounts
Changed geberation of sensor unique id (breaking change)

* Fixed lints

* Fixed error detecting opened window

* Changed gereration of unique id of climate

* Removed commented code and added comments

balloob committed:
Disable Hue groups for new setups (#31713)

bdraco committed:
Significantly reduce the number of API calls that the august integration (#31685)

* Significantly reduce the number of API calls that the august integration
makes.

The poll interval for the lock status API is now 15 minutes
instead of every 10 seconds because we can use the activity
API to see changes in lock state.  The interval for the
activity API is 10 seconds which allows for the same
frequency of state monitoring without all the additional API calls.
With four locks, this change results in an ~80% reduction in the number
of API calls.

The result of the lock and unlock APIs now update the lock state instead
of waiting for the next poll.  This change also has the added benefit of
making the UI appear far more responsive.

* Convert to using UTC times

balloob committed:
Entity Registry to store and restore name/icon (#31714)

* Entity Registry to store and restore name/icon

* Update test_entity_registry.py

* Add original name/icon to JSON result

kit-klein committed:
Support Konnected Pro alarm panel, embrace async, leverage latest HA features/architecture (#30894)

* fix unique_id computation for switches

* update konnected component to use async, config entries, registries. Pro board support and tests

* clean up formatting comments from PR

* use standard interfaces in tests

* migrate config flow to use options

* address latest pr feedback

* format for import as part of config schema validation

* address pr feedback

* lint fix

* simplify check based on pr feedback

* clarify default schema validation

* fix other schema checks

* fix translations

Co-authored-by: Nate Clark <nate@nateclark.com>

maxcanna committed:
Fix set volume level (#31731)

afsapi requires an `int` in the 0-20 range but we receive a `float` in the 0.0-1.0 range so we have to cast and offset it

vzahradnik committed:
Update Modbus service manifest (#31727)

dgwynne commented:
How do I push this forward again?

homeassistant committed:
[ci skip] Translation update

dgomes committed:
pyipma version bump (#31739)

bdraco committed:
Read door open/close events from the activity log. (#31732)

As polling for lock status is every 15 minutes,
we read lock and unlock events from the activity log.

An upstream update of py-august was needed to
expose the door open and close events.

Door open and close events are now seen within
a few seconds instead of delayed 15+ minutes.

scop committed:
Improve Huawei LTE timeouting/stalling request behavior (#31710)

* Suppress data get timeout exceptions for 30s after notify attempts

At least SMS send failures may put the API in a state where it times out
some (but not all) data get operations for some time, e.g. 25s.

Closes https://github.com/home-assistant/home-assistant/issues/30827

* Do not pile up duplicate data requests

Do not add another request for a piece of data for which a previous
request is still in progress. For example failing SMS sends are known to
stall some (but not all) requests for some time, and firing up more is
not going to help.

cgtobi committed:
Fix missing device class in netatmo binary sensors (#31693)

* Bring back device class

* Add door tag sensors types

* Actually discover individual tags per camera

cgtobi committed:
Fix missing device class in netatmo binary sensors (#31693)

* Bring back device class

* Add door tag sensors types

* Actually discover individual tags per camera

bdraco committed:
Refresh the august access token when needed (#31735)

* Refresh the august access token
when needed.

Currently august will stop working when the token
expires about every six month.

This resolves issue #23788

* Make refresh_access_token_if_needed private since we do not want additional callers

* Add init

robbiet480 committed:
Add new webhook action to allow enabling encryption in an exis… (#31743)

* Add new webhook action to allow enabling encryption in an existing registration

* Harden tests

* Make requested fixes

balloob committed:
Use set for dependency lookup in hassfest (#31746)

frenck committed:
Fix spelling of ecobee in manifest (#31751)

vzahradnik committed:
Add availability status to Modbus entities and improve error handling (#31073)

basnijholt committed:
Fix moving average test for discrete derivative sensor (#31750)

* fix test_data_moving_average_for_discrete_sensor

After https://github.com/home-assistant/home-assistant/pull/31717 the test
didn't actually test anything anymore.

This fixes that.

* make the test faster

jjlawren committed:
Improve Plex media_player entity naming (#31755)

ManneW committed:
Add initial version of Vilfo Router integration (#31177)

* Initial implementation of Vilfo router integration.

This commit is a combination of several commits, with commit messages in bullet form below.

* Added additional files to Vilfo integration.
* Added generated files.
* Fixed alphabetic order in generated config_flows.
* Continued implementation of config flow for Vilfo integration.
* Continued work on config_flow for Vilfo.
* Updated requirements in manifest for Vilfo Router integration.
* Some strings added to Vilfo Router integration.
* Vilfo Router integration updated with sensor support.
* Code style cleanup.
* Additional cleanup of config flow.
* Added additional UI strings for Vilfo Router
* Updated tests of config flow and fixed formatting
* Updated requirement upon vilfo-api-client.
* Sensor refactoring including support for icons
* Code style changes for Vilfo Router integration
* Code cleanup
* Fixed linting issues in Vilfo Router integration
* Fixed import order in test for Vilfo integration.

* Updates to Vilfo Router integration based on feedback

Based on the feedback received, updates have been made to the Vilfo Router integration.

A couple of the points mentioned have not been addressed yet, since the appropriate action has not yet been determined. These are:

* https://github.com/home-assistant/home-assistant/pull/31177#discussion_r371124477
* https://github.com/home-assistant/home-assistant/pull/31177#discussion_r371202896

This commit consists of:

    * Removed unused folder/submodule
    * Fixes to __init__
    * Fixes to config_flow
    * Fixes to const
    * Refactored sensors and applied fixes
    * Fix issue with wrong exception type in config flow
    * Updated tests for Vilfo integration config_flow
    * Updated dependency upon vilfo-api-client to improve testability
    * Import order fixes in test
    * Use constants instead of strings in tests

* Updated the VilfoRouterData class to only use the hostname as unique_id when it is the default one (admin.vilfo.com).

* Refactored based on feedback during review.

* Changes to constant names,
* Blocking IO separated to executor job,
* Data for uptime sensor changed from being computed to being a timestamp,
* Started refactoring uptime sensor in terms of naming and unit.
* Updated constants for boot time (previously uptime) sensor.
* Refactored test of Vilfo config flow to avoid patching code under test.
* UI naming fixes and better exception handling.

* Removed unused exception class.

* Various changes to Vilfo Router integration.

* Removed unit of measurement for boot time sensor,
* Added support for a sensor not having a unit,
* Updated the config_flow to handle when the integration is already configured,
* Updated tests to avoid mocking the code under test and also to cover the aforementioned changes.

* Exception handling in Vilfo Router config flow refactored to be more readable.

* Refactored constant usage, fixed sensor availability and fix API client library doing I/O in async context.

* Updated signature with hass first

* Update call to constructor with changed order of arguments

Commit:
Fix smoke detection for HomematicIP Cloud (#31753)

Commit:
Fix smoke detection for HomematicIP Cloud (#31753)

balloob committed:
Whitelist shopping list updated event (#31742)

* Whitelist shopping list updated event

* Add ignore to hassfest

dshokouhi committed:
Add device name to sensor name for mobile_app (#31756)

* Add device name to sensor name

* Update test to include device name

Adminiuga committed:
ZHA tests refactoring (#31744)

* Refactor ZHA fixtures.
Patch Zigpy radio libs instead of ZHA when setting up fixtures.
Use new fixtures for binary_sensor.zha platform.

* Update ZHA api tests.
* Update ZHA channels and discovery tests.
* Update ZHA cover tests.
* Update device action/trigger tests.
* Update device_tracker.zha platform tests.
* Update fan.zha platform tests.
* Update ZHA gateway tests.
* Update lock.zha platform tests.
* Update switch.zha platform tests.
* Update sensor.zha platform tests.
* Update light.zha platform tests.
* Use MockConfigEntry.
* Address PR comments.

bramkragten committed:
Updated frontend to 20200130.3 (#31771)

bramkragten committed:
Updated frontend to 20200130.3 (#31771)

bramkragten committed:
Updated frontend to 20200212.0 (#31772)

bdraco committed:
Cleanup August activity processing and add tests (#31774)

* Update py-august to 0.12.0

* Upstream update also resolves issue #28960

frenck committed:
Fix spelling of VIVOTEK (#31773)

* Fix spelling of VIVOTEK in manifest

* Also adjust default name of camera to match the brand name

frenck committed:
Fix spelling of apcupsd in manifest (#31770)

frenck committed:
Fix spelling of AVM FRITZ!Box in manifest (#31765)

Quentame committed:
Implement PlatformNotReady to Linky + fix TypeError (#31768)

Kane610 committed:
UniFi - Change handling of updated options (#31762)

* Change handling of updated options

* Add tests

balloob committed:
Rename codecov so it will be picked up (#31775)

homeassistant committed:
[ci skip] Translation update

frenck committed:
Fix spelling of ASUSWRT in manifest (#31764)

* Fix spelling of ASUSWRT in manifest

* Update homeassistant/components/asuswrt/__init__.py

Co-Authored-By: springstan <46536646+springstan@users.noreply.github.com>

Co-authored-by: springstan <46536646+springstan@users.noreply.github.com>

dingusdk committed:
Update to version 2.6 of ihcsdk (#31789)

* Make the set_runtime_value_int function work with template values

* Use newest version of the ihcsdk library

* Make the set_runtime_value_int function work with template values

* Use newest version of the ihcsdk library

* Updated to the newest ihcsdk 2.5.0

* Formatted changes to make it pass CI tests

* Update to version 2.6 of ihcsdk

frenck committed:
Spelling: Config(uration) (#31782)

scop committed:
Add and use bunch of data size and rate related constants (#31781)

Also fix a few units to match the corresponding data.

frenck committed:
Fix spelling of NETGEAR and CalDAV in manifests (#31790)

balloob committed:
Write state if schedule update state from async context (#31758)

* Write state if schedule update state from async context

* Fix most tests

* Fix test and PS4 I/O in event loop

* Fix ps4 better

bachya committed:
Add support for real-time data from SimpliSafe (#31424)

* Add support for real-time data from SimpliSafe

* Updated requirements

* Linting

* Ensure dispatcher topic contains the domain

* Don't bother with a partial

* Websovket dataclass and other code review

* Ensure initial_event_to_use works with error

* Don't inline methods

* Don't abuse loop variable

* Simplify initial event retrieval

* Add connection lost and restored events

* Revert "Add connection lost and restored events"

This reverts commit e7ffe05938e6cd13a5426f8a605260056fa04de0.

* Make _on_disconnect a static method

* Code review comments

* Allow entities to opt out of REST and/or websocket API updates

* Revert "Allow entities to opt out of REST and/or websocket API updates"

This reverts commit 1989f2e00e0b95dd466bcc803e7c83afab6d2763.

* Code review comments

* Fix issues with events not triggering correct entities

* Bug fixes

MrDadoo committed:
Update onewire component (#31419)

* Added some measurement points for Family 26.
Added three attriubtes for each sensor, DEVICE_FILE, RAW_VALUE and SENSOR_TYPE
Added some comments.
Updated to get config values for owserver to get owport and owhost.

* Changed to _LOGGER.debug in some places

Resorted includes

* Fixup of code to reflect review comment, comply to Black and pass tests

* Added unique_id Entity property with device file id and added device_file as attribute that takes its data from Entities unique_id-.

* Missing blank line identified by fake8 and black

* Changed to let device_state_attributes return attribute device_file from the self._device_file member variable of the entity.

* Changed from info to debug logging

Kane610 committed:
Add multi select support to config validation and to custom serializer (#31798)

balloob committed:
Catch unexpected exceptions when validating config (#31795)

scop committed:
Use time.monotonic instead of time.time where appropriate (#31780)

robbiet480 committed:
Add Home Assistant Companion to manifest.json so we can sugges… (#31808)

jjlawren committed:
Update Plex connection class to push (#31806)

balloob committed:
Update codecov.yml

Commit:
Remove force from async_schedule_update_ha_state for HMIPC (#31796)

* remove force from async_schedule_update_ha_state if HMIPC

* Fix lint

homeassistant committed:
[ci skip] Translation update

jjlawren committed:
Add summary attribtue for currently playing media (#31803)

vzahradnik committed:
Fix swap of min and max default values in Modbus climate (#31801)

jjlawren committed:
Add device registry support for Plex (#31797)

* Add device registry support for Plex

* Fix model

* Use Plex server as sensor device identifier

ctalkington committed:
update directv to directpy==0.6 (#31812)

Commit:
Speed up tests of HomematicIP Cloud (#31810)

* Speed up tests of HomematicIP Cloud

* load the json once

abmantis committed:
Allow hourly forecast in IPMA (#30979)

* update ipma component for pyipma 2.0
* fix wind speed; refactor forecast
* update requirements*.txt
* fix tests; update CODEOWNERS; update pyipma to 2.0.1
* minor changes as suggested in PR
* make lint happy
* fix mocking coroutines
* restore old unique id
* fix station lat/lon; update pyipma version
* add hourly forecast option to IPMA
* add forecast tests
* use for instead of lambda

balloob committed:
Limit translations_develop to an integration (#31804)

* limit translations_develop to english

* Convert to Python

* Limit to integration

* Add to hassfest

* Remove old bash comment

balloob committed:
Annotate more async functions correctly (#31802)

bachya committed:
Fire HASS events on SimpliSafe events (#31811)

* Fire HASS events on SimpliSafe events

* Bump simplisafe-ptyhon to 7.3.0

* Update reqirements

* Updates

* Revert "Updates"

This reverts commit 55818894176fe27938c79685ab87100d1bc120ff.

* Restrict which events get fired

* Code review comments

Kane610 committed:
Change multi_select config validator to class (#31828)

* Move multi_select to class

* Fix serializer and add test

* Serializer should also return options

Commit:
Remove energy sensor from incompatible Ata devices (#31831)

An AtaDevice has a boolean flag describing whether it supports energy
consumption metering. The flag was ignored resulting in sensor entities
reporting constant 0 kWh consumption.

* Update pymelcloud dependency to support the has_energy_consumed_meter
flag.

* Add ATTR_ENABLED_FN to sensor definitions for filtering out
unsupported sensors.

* Fix typing issue in sensor constructor.

* Remove unused UnitSystem constructor parameter.

frenck committed:
Spotify integration hotfixes (#31835)

* Remove services file, incorrect info

* Guard currently playing for being a NoneType

* Revert "Guard currently playing for being a NoneType"

This reverts commit f5f56b0db03b407e058d45cd3549af1388916e06.

* Guard currently playing item is None

* Process review suggestions

frenck committed:
Spotify integration hotfixes (#31835)

* Remove services file, incorrect info

* Guard currently playing for being a NoneType

* Revert "Guard currently playing for being a NoneType"

This reverts commit f5f56b0db03b407e058d45cd3549af1388916e06.

* Guard currently playing item is None

* Process review suggestions

balloob committed:
Fix person device_trackers null (#31829)

balloob committed:
Fix person device_trackers null (#31829)

balloob committed:
Google Assistant: Remove speaker type and earlier filter out devices from being locally exposed (#31830)

* Remove speaker type

* Do not expose locks or alarms to Google Local

balloob committed:
Google Assistant: Remove speaker type and earlier filter out devices from being locally exposed (#31830)

* Remove speaker type

* Do not expose locks or alarms to Google Local

balloob committed:
Bumped version to 0.105.4

balloob committed:
Revert "For vizio integration, set unique ID early to prevent multiple zeroconf discovery items for the same device to appear (#31686)"

This reverts commit 295963f8e8c554bf8b614dae06aca95ccc99ade6.

balloob committed:
Merge pull request #31841 from home-assistant/rc

0.105.4

homeassistant committed:
[ci skip] Translation update

maxcanna committed:
Add turn_on_action configuration variable (#31792)

Allow to turn on the TV using an external service

robmarkcole committed:
Bump pillow to 7.0 (#31847)

scop committed:
Simplify missing Garmin Connect data handling, mark entities un/available (#31718)

* Simplify missing Garmin Connect data handling, mark entities un/available

* Remove unnecessary else

michaelarnauts committed:
Add Tado water_heater (#30095)

* Add Tado water_heater

* Don't use climate CONSTS

* Fix logging text

* Add changes for multiple bridge support

* Address remarks

* should_poll must be False

* Remove additional async_schedule_update_ha_state()

* Not for climate

MartinHjelmare committed:
Revert "Check netgear device_tracker link_rate to ensure device is connected" (#31855)

This reverts commit 669844e4ddc7c3733ef03e4baf0e3ba6d1c5c3a6.

scop committed:
Helpers typing improvements (#31865)

balloob committed:
Update codecov.yml

maxcanna committed:
Improve media name detection on lg_netcast (#31863)

When the tv is using an external input (e.g. HDMI1) values have to be taken from fields otherwise they'll empty

Danielhiversen committed:
update pyTibber libary, add signal strength for Pulse and watty (#31851)

* update pyTibber libary, add signal strength for Pulse and watty

* update pyTibber libary, add signal strength for Pulse and watty

caronc committed:
Bump Apprise version to v0.8.4 (#31868)

moshekaplan committed:
Add support for displaying Daf Yomi (#30628)

* Add support for displaying Daf Yomi

* Ran black --fast

* Added docstring to get_daf

* Further lint fixes

* Remove unnecessary else

* clarify code

* Use fstrings

* pull daf yomi from hdate

* Update manifest version for daf_yomi support

* fix variable usage

* Update requirements

* Also pass in today's date as well

* Rename date variable to daytime_date

* Add tests for daf yomi sensor

* Update stale test IDs

jjlawren committed:
Use new custom_serializer (#31871)

homeassistant committed:
[ci skip] Translation update

PhilRW committed:
Fix CalDAV recurring events (#31805)

* Fix CalDAV parsing of recurring events

Some CaDAV servers (see: SOGo) return the original event that contains
the recurrence rules. The CalDAV calendar component sorts and filters
events based on their start and end dates, and was failing to properly
show recurring events based on these recurrence rules.

This this change checks if an event has recurrence rules and changes the
start/end dates of the event to today if the event is set to occur
today. This allows the rest of the component logic to function properly.

* Use date from nextmost occurence

* Adding unit tests

* Add endless event unit test

* Create new vevent for each event recurrence today

* Remove redundant unit test

* Add timezone to events that have none

Python cannot compare them otherwise.

* Simplify code, add comments & guard clause

* Add test for recurring all day event

* Account for all-day events

* Remove redundant code

* Remove redundant code

* Remove unnecessary deepcopy

* Add hourly recurring tests

* Add tests for hourly repeating event

* Fix unit test

* Use event.copy()

bdraco committed:
Convert august to async so a token refresh lock can be used (#31848)

* Convert august to async so a token refresh lock can be used

* Update comment since we now have a lock

* Do not mock the lock

* Address review items

jcalbert committed:
Add rainforest_eagle support for legacy hardware (#28082)

* Add compatibility with Legacy EAGLE models.

* added mdns resultion via zeroconf

* Added requirements to requirements_all.txt

* Fixed model determination bug

* Fixed formatting issue for Eagle-200 responses.

* Remove mDNS resolution, IP now required.

* added pypi deps, fixed handling of optional params

manifest updates

* Fixed import order per isort.

* Two pylint fixes.

* More consistent try-fail structure to guessing hardware.  Errors actually fail.

bdraco committed:
Support XML conversion for RESTful sensors (#31809)

* Support XML conversion for RESTful sensors

Many devices continue to use XML for RESTful
APIs.  Interfacing with these APIs requires custom
integrations or command line fork()/exec() overhead
which many of these devices can work with as if
they were JSON using xmltojson via this spec:
https://www.xml.com/pub/a/2006/05/31/converting-between-xml-and-json.html

This change implements converting XML output to
JSON via xmltojson so it can work with the existing
rest sensor component.  As the attributes that
usually need to be scraped are deeper in the document
support for passing in a template to find the
JSON attributes that have been added.  JSON APIs that
do not have their attributes at the top level
can also benefit from this change.

* Auto convert xml, change out the template for jsonpath

* Address review items and potentially unexpected normalize behavior with jsonpath

* Revert "Address review items and potentially unexpected normalize behavior with jsonpath"

This reverts commit fe9e179092f2a664e3668e0c832856bf5c78d262.

* json_dict[0] turned out to be needed

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

Marco98 committed:
Fix mikrotik detecting capsman support (#31819)

* fix detecting capsman support

* fix isort

* moved support_capsman to update_devices()

* moved support-checks to setup method

* moved setup method to get_hub_details

* implement suggestion for device lists

* fix black formatting

* remove not needed variable wireless_devices

* fix usage of force_dhcp to seperate wireless devs

* fix black fmt

Commit:
Adjust tests after speed up to restore coverage for HomematicIP Cloud (#31836)

* Adjust tests after speed up to restore coverage for HomematicIP Cloud

* Fix test data

* Fixes after review

* remove duplicate cade

* remove service marker

cereal2nd committed:
Upgrade python-velbus to 2.0.41 (#31875)

scop committed:
Type hint improvements (#31876)

* Complete components.remote type hints

* Define ConfigType only in helpers.typing

jardiamj committed:
Bump adafruit-blinka and adafruit-circuitpython-mcp230xx (#31845)

* Updated MCP23017 component to use latest adafruit-blinka v-3.9.0 and Adafruit_CircuitPython_MCP230xx v-2.2.2

* Added updated dependencies to requirements_all.txt

Squixx committed:
Update nederlandse_spoorwegen to properly handle punctuality (#31741)

laszlojakab committed:
Add timestamp to lg_netcast media_image_url to update image correctly (#30933)

* add timestamp to media_image_url to update image correctly

* applying isort on source

* apply black formatting on source

* using f strings in media_image_url property

* remove unnecessary casting

MartinHjelmare committed:
Clean up netgear device tracker (#31861)

* Improve logging

* Clean up login

* Clean docstring

* Clean config access

* Remove default None for port

* Add not working guard

* Remove debug print

frenck committed:
Activate Stale bot for PR's (#31837)

* Activate Stale bot for PR's

* Disable issues only

gerard33 committed:
Upgrade bimmer_connected to 0.7.0 (#31894)

gerard33 committed:
Upgrade bimmer_connected to 0.7.0 (#31894)

MartinHjelmare committed:
Fix check_real location guard (#31890)

MartinHjelmare committed:
Clean soundtouch (#31888)

* Clean up soundtouch tests

* Remove not needed updates

homeassistant committed:
[ci skip] Translation update

jkeljo committed:
Fix bugs in greeneye_monitor voltage reporting (#31895)

* Fix #31870

* Fix voltage notifications too

jkeljo committed:
Upgrade greeneye_monitor to 2.0 (#31896)

mueslo committed:
prevent dev_id being permanently assigned as config_name (#31886)

BKPepe committed:
Upgrade youtube_dl to version 2020.02.16 (#31905)

balloob committed:
Upgrade aiohue (#31903)

mezz64 committed:
Add POD support, sleep fitness sensor to EightSleep (#31874)

* Add POD support, sleep fitness sensor

* Black format changes

* Change dict formatting

balloob committed:
Add logger name to system log (#31902)

balloob committed:
Allow specifying an icon for a scene (#31898)

balloob committed:
Add icons to scripts (#31899)

Commit:
Bump dependency to 0.10.17 for HomematicIP Cloud (#31911)

balloob committed:
Start safe mode if invalid core conf (#31904)

guimaraes13 committed:
Hue Group features based on the bulbs in it (#31897)

* Computes the features of a hue group as the union of the features of the bulbs in the group

* Moved create item to a function

* Added test for hue group features

bramkragten committed:
Updated frontend to 20200212.0 (#31912)

bdraco committed:
Provide user consumable errors when lock operations fail (#31864)

* Provide user consumable errors when lock operations fail

This resolves issue #26672

* include from in raise

* pylint

* Cleanup of mocking.

balloob committed:
Report which data causes JSON serialization error (#31901)

balloob committed:
Report which data causes JSON serialization error (#31901)

bachya committed:
Bump simplisafe-python to 8.1.1 (#31915)

frenck committed:
Remove duke_energy integration (ADR-0004) (#31921)

bramkragten committed:
Updated frontend to 20200217.0 (#31922)

Quentame committed:
Add async_unload_entry to iCloud (#31917)

bramkragten committed:
Add an options flow to demo (#31920)

To help test the frontend

balloob committed:
Bumped version to 0.105.5

balloob committed:
Merge pull request #31931 from home-assistant/rc

0.105.5

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

exxamalte committed:
Bump aio_geojson_geonetnz_quakes to 0.12 (#31930)

homeassistant committed:
[ci skip] Translation update

austinmroczek committed:
Fix translations_develop (#31942)

exxamalte committed:
Bump aio_geojson_nsw_rfs_incidents to 0.3 (#31941)

ktnrg45 committed:
Bump pyps4-2ndscreen to 1.0.7 (#31943)

Poeschl committed:
Add pigpio remote functionality (#31667)

balloob committed:
Fix entity registry not saving name/icon (#31932)

balloob committed:
Untag discovery as an allowed dependency (#31934)

vzahradnik committed:
Read Modbus climate current temp from input register (#31944)

bachya committed:
Add support for SimpliSafe system notifications (#31936)

* Add support for SimpliSafe system notifications

* Actually process notifications

* Code review comments

Poeschl committed:
Add Twitch subscription and additional stats (#31122)

* add oauth functionality and additional attributes

* Add tests WIP

* Make mocks work the correct way

* Use CONF_TOKEN constant for config

* Remove twitch from .coveragerc

* Update homeassistant/components/twitch/sensor.py

Lets be consistent

Co-Authored-By: springstan <46536646+springstan@users.noreply.github.com>

Co-authored-by: springstan <46536646+springstan@users.noreply.github.com>

balloob committed:
Extend safe mode (#31927)

* Extend safe mode

* Add safe mode boolean to config JSON output and default Lovelace

* Add safe mode to frontend

* Update accent color

balloob committed:
Do not use a set for hvac modes in SmartThings (#31959)

bdraco committed:
Audit state handling off august bridges and sensors (#31935)

* Audit state handling of august bridges and sensors

This addresses issue #29980

* Prevent setting up august locks that do not have a bridge as they will never work

* Prevent locks showing available when their bridge is offline

* Prevent door sensors from showing available when their bridge is offline

* Prevent creating door sensors for locks that do not have them

* Prevent doorbells showing unavailable when they are in standby mode

* Set SCAN_INTERVAL for binary_sensors to 5 seconds as
  data comes in from the activity endpoint more frequently

* Update homeassistant/components/august/__init__.py

raise if the detail is missing when checking doorsense

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* Handle another place where the lock detail could not exist

* Address review comments

* Handle lock detail update failing and add test

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

Kane610 committed:
deCONZ - Directly reflect changes to config entry options (#31661)

* Directly reflect changes to config entry options

* Remove print

* Add tests

* Add title to config options

frenck committed:
Upgrade pre-commit to 2.1.0 (#31962)

teharris1 committed:
Fix pre-commit hook issue on Windows (#31648)

* Make shebang compatable with Windows

* Prehook test

* Undo prehook test

emontnemery committed:
Add support for MQTT device triggers (#31679)

* Add support for MQTT device triggers

* Fix test, tweaks

* Improve test coverage

* Address review comments, improve tests

* Tidy up exception handling

* Fix abbreviations

* Rewrite to handle update of attached triggers

* Update abbreviation test

* Refactor according to review comments

* Refactor according to review comments

* Improve trigger removal

* Further refactoring

Kane610 committed:
UniFi config option SSID filter (#31842)

* Draft

* Use new multi_select config validation

* Bump dependency to v13

* Improve options flow

* Add title to config options

* Add config option descriptions

* Fix martins comment

Kane610 committed:
deConz - Use proper mechanisms for options flow tests (#31965)

Generic clean up

balloob committed:
Optimize find bad JSON data (#31963)

homeassistant committed:
[ci skip] Translation update

jjlawren committed:
Add options to ignore shared/managed Plex clients (#31738)

* Add option to ignore shared/managed Plex clients

* Start to allow user selection

* hass object not ready during init

* Don't bother sorting

* Plex account multi-select, handle new users not matching config

* Fix/add tests

* Lint simplifications

* Review cleanup

* Oops

* Rename options attributes, fix/add tests

jjlawren committed:
Fix Plex naming and devices for misbehaving clients (#31970)

gerard33 committed:
Upgrade to bimmer_connected 0.7.1 to fix US and China access (#31968)

vzahradnik committed:
Modbus remove unnecessary get calls (#31966)

springstan committed:
Bump vallox-websocket-api to 2.4.0 (#31913)

SukramJ committed:
Add new devices to HomematicIP Cloud (#31914)

* Add new devices to HomematicIP Cloud

Adds:
- HMIP-WTH-B (Wall Thermostat Basic)
- ALPHA-IP-RBG (Alpha IP Wall Thermostat Display)
- ALPHA-IP-RBGa (ALpha IP Wall Thermostat Display analog)

* Update homeassistant/components/homematicip_cloud/sensor.py

Co-Authored-By: springstan <46536646+springstan@users.noreply.github.com>

Co-authored-by: springstan <46536646+springstan@users.noreply.github.com>

SukramJ committed:
Add shutter group to HomematicIP Cloud (#31916)

SukramJ committed:
Fix test for HMIPC (#31981)

bramkragten committed:
Updated frontend to 20200219.0 (#31979)

dmulcahey committed:
Bump ZHA quirks and add skip configuration support (#31982)

* add skip configuration
* Bump quirks
* add skip configuration to FakeDevice

balloob committed:
Only check frontend for safe mode if frontend wanted to be loa… (#31969)

* Only check frontend for safe mode if frontend wanted to be loaded

* Update test

pvizeli committed:
Nuki: add support for unique id (#31824)

* Nuki support unique id and the battery level attribute

* Fix isort

* Address comments

* Cache attribute

* Cleanup

* Restore false

* Fix isort

homeassistant committed:
[ci skip] Translation update

marthoc committed:
Fix bug in ecobee integration (#32008)

* Bump python-ecobee-api to 0.2.1

* Update log messages for clarity

* Update requirements_all

tsvi committed:
Don't return coroutine in DLNA/DMR service handler (#32011)

Fixes #32010

danielperna84 committed:
Update pyhomematic to 0.1.65 (#32006)

jjlawren committed:
Fix Plex sensor title handling (#31973)

* Fix Plex sensor title parsing

* Revert episode year for now

frenck committed:
Upgrade requests to 2.23.0 (#32013)

frenck committed:
Bumped version to 0.107.0dev0

dgwynne committed:
provide an abbreviation for "speed_command_template".

hopefully this fixes the CI tests

dgwynne committed:
provide an abbreviation for "speed_command_template".

hopefully this fixes the CI tests

frenck commented:
@dgwynne There is nothing to do for you to push it forward, this PR now awaits review.

frenck committed:
Deprecate Xfinity Gateway integration (ADR-0004) (#32017)

MatthewFlamm committed:
Add missing name to logging in DataUpdateCoordinator (#32023)

balloob committed:
Fix hue test

frenck committed:
Remove deprecated Hue options (fixes CI) (#32027)

cgtobi committed:
Bump pyatmo to 3.2.4 (#32018)

* Bump pyatmo to 3.2.3

* Bump pyatmo version to 3.2.4

* Update requirements_test_all.txt

frenck committed:
Remove states UI options in group integration (#32021)

balloob committed:
Add entity ID to input_number warning (#32012)

* Add entity ID to warning

* Input Number flexibility

raman325 committed:
Update vizio dependency and integration name (#31975)

* bump pyvizio version and clean up code

* fix requirements

* update parameter to optional per review in docs PR

robmarkcole committed:
Add convert_to_pil_image to pil util (#31825)

* Update pil.py

* Update doods and tensorflow to use convert_to_pil_image

* Update pil.py

* Add log messages on bad data

* Drop convert_to_pil_image

Just perform conversion in the integrations without seperate convert_to_pil_image()

frenck committed:
Remove weblink integration (#32024)

balloob committed:
Fix recursion bug (#32009)

* Fix recursion bug

* Remove shield

frenck committed:
Remove history_graph integration (#32028)

* Remove history_graph integration

* Update codeowners file

frenck committed:
Prevent committing to dev/master/rc directly (#32029)

* Prevent committing to dev/master/rc directly

* Skip pre-commit on version bump

frenck committed:
Add minimal version contrain to urllib3 (#32031)

balloob committed:
Updated frontend to 20200220.0 (#32033)

frenck committed:
Fix extra arguments of group integration (#32032)

balloob committed:
Template platform tweaks (#32037)

* Remove unnecessary boolean return from platform setup

* Fix template cover validation

robbiet480 committed:
Change TTS codeowner to @pvizeli (#32041)

balloob committed:
Updated frontend to 20200220.1 (#32046)

springstan committed:
Update file headers and outdated documentation links (#32022)

* Update documentation links

* Update file headers v2

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Significantly reduce code in august integration (#32030)

* Significantly reduce code in august integration

* Activity updates can now be processed by py-august
  this allows us to eliminate the activity sync
  code for the door sensors and locks

* Lock and door state can now be consumed from
  the lock detail api which allows us to
  remove the status call apis and reduce
  the number of API calls to august

* Refactor the testing method for locks (part #1)

* Update homeassistant/components/august/binary_sensor.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* Switch to asynctest instead of unittest for mock.patch

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

dgwynne committed:
Merge branch 'mqtt.fan.speed_cmd_tpl' of dgwynne/home-assistant into mqtt.fan.speed_cmd_tpl

homeassistant commented:
Hi @SoftXperience, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#31655).

Once you do that we will be able to review and accept this pull request.

Thanks!

MartinHjelmare commented:
The branch isn't clean anymore. Please either rebase and clean up the branch or open a new PR from a fresh clean branch.

dgwynne commented:
Understood. Closing this PR and trying again.
